staging/src/k8s.io/apiserver/pkg/endpoints/handlers/watch.go

```diff
@@ -173,13 +173,12 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		return
 	}
 
-	cn, ok := w.(http.CloseNotifier)
-	if !ok {
-		err := fmt.Errorf("unable to start watch - can't get http.CloseNotifier: %#v", w)
-		utilruntime.HandleError(err)
-		s.Scope.err(errors.NewInternalError(err), w, req)
-		return
-	}
+	// Use context to check client closed instead of http.CloseNotifier.
+	// Since http.CloseNotifier is deprecated, what's worse,
+	// http.CloseNotifier will cost 1 more goruntine if proto is HTTP/2.x.
```

Comment:
A lot of this comment belongs in the commit message, not in the code. The code comment could be `// Use context instead of CloseNotifier as it's more efficient`, or simply nothing as there's not anything particularly tricky about this.

Reply:
yes, I have removed comment. Thanks!